cmd/egress-cni-plugin/egressContext.go

```diff
@@ -233,101 +239,426 @@ func (c *EgressContext) SetupHostVeth(vethName string) error {
+
+	containerIPv6, err := ec.getContainerIpv6GlobalAddrs(containerInterface.Name)
+	if err != nil {
+		ec.Log.Errorf("container interface %s failed to obtain a unique-local IPv6 address",
+			containerInterface.Name)
+		return err
+	}
+	if len(containerIPv6) > 1 {
+		ec.Log.Warnf("more than one IPv6 global unicast address obtained, ifName: %s, IPs: %s, %s is selected for IPv6 egress traffic",
+			containerInterface.Name, containerIPv6, containerIPv6[0].String())
+	} else if len(containerIPv6) < 1 {
+		ec.Log.Errorf("no IPv6 global unicast address obtained, ifName: %s, IPs: %s",
+			containerInterface.Name, containerIPv6)
+		return fmt.Errorf("no IPv6 global unicast address found, ifName: %s, IPs: %s",
+			containerInterface.Name, containerIPv6)
+	}
+
+	err = ec.setupContainerIPv6Route(hostInterface, containerInterface)
+	if err != nil {
+		ec.Log.Errorf("setupContainerIPv6Route failed: %v", err)
+		return err
+	}
+	ec.Log.Debugf("container route set up successfully")
+
+	err = ec.setupHostIPv6Route(hostInterface, containerIPv6[0])
+	if err != nil {
+		ec.Log.Errorf("setupHostIPv6Route failed: %v", err)
+		return err
+	}
+	ec.Log.Debugf("host IPv6 route set up successfully")
+
+	// set up SNAT in host for container IPv6 egress traffic
+	// following line adds an ip6tables entries to NAT for IPv6 traffic between container v6if0 and node primary ENI (eth0)
+	err = snat.Add(ec.IPTablesIface, ec.NetConf.NodeIP, containerIPv6[0], ipv6MulticastRange, ec.SnatChain, ec.SnatComment, ec.NetConf.RandomizeSNAT)
+	if err != nil {
+		ec.Log.Errorf("setup host snat failed: %v", err)
+		return err
+	}
+
+	ec.Log.Debugf("host IPv6 SNAT set up successfully")
+
+	mergeResult(ec.Result, ec.TmpResult)
+	// remove comment for debug purpose
+	// ec.Log.Debugf("output result: %+v", *ec.Result)
+
+	// Pass through the previous result
+	return types.PrintResult(ec.Result, ec.NetConf.CNIVersion)
+}
+
+// cmdDelEgressV6 exec clear the setting to support IPv6 egress traffic in EKS IPv4 cluster
+func (ec *egressContext) cmdDelEgressV6() (err error) {
+	var contIPAddrs []netlink.Addr
+
+	if ec.IPTablesIface == nil {
+		if ec.IPTablesIface, err = ec.IptCreator(iptables.ProtocolIPv6); err != nil {
+			ec.Log.Error("command ip6tables not found")
+		}
+	}
+
+	if ec.NsPath != "" {
+		err = ec.Ns.WithNetNSPath(ec.NsPath, func(hostNS ns.NetNS) error {
+			var containerLink netlink.Link
+			var _err error
+			ec.Log.Debugf("trying to get link by name %s", ec.NetConf.IfName)
+			containerLink, _err = ec.Link.LinkByName(ec.NetConf.IfName)
+			if _err != nil {
+				if !cniutils.IsLinkNotFoundError(_err) {
+					ec.Log.Debugf("failed to get link by name %s: %v", ec.NetConf.IfName, _err)
+				}
+				// must return when err, following code needs containerLink
+				return _err
 			}
 
-			if err != nil && err == ip.ErrLinkNotFound {
-				c.Log.Debugf("DEL: Link Not Found, returning", err)
-				return nil
+			contIPAddrs, _err = ec.Link.AddrList(containerLink, netlink.FAMILY_V6)
```

Comment:
Is the container IPv6 address available in PrevResult? I think we might be able to get it from `ec.NetConfig.PrevResult`

Reply:
the updated version combined v4 del and v6 del, and removed v6 mergeResult, so the prevResult will not have ip address anymore. The v4 did not have ip stored in result in our original code.